File: 250624/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.java
import java.util.*;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        Person[] people = new Person[n];
        for (int i = 0; i < n; i++) {
            String name = sc.next();
            int height = sc.nextInt();
            int weight = sc.nextInt();
            // Please write your code here.
            people[i] = new Person(name, height, weight);
        }
        Arrays.sort(people, (a,b) -> {
            if (a.height == b.height) {
                return b.weight - a.weight;
            }
            return a.height - b.height;
        });

        for (int i = 0; i < n; i++) 
            System.out.println(people[i].name + " " + people[i].height + " " + people[i].weight);
    }
}

class Person {
    String name;
    int height;
    int weight;

    Person(String name, int height, int weight) {
        this.name = name;
        this.height = height;
        this.weight = weight;
    }
}

File: 250925/키, 몸무게를 기준으로 정렬/sort-by-height-and-weight.py
n = int(input())
name = []
height = []
weight = []

arr = []

class S :
    def __init__(self,n,h,w) :
        self.n = n
        self.h = h
        self.w = w
        
for _ in range(n):
    n_i, h_i, w_i = input().split()
    name.append(n_i)
    height.append(int(h_i))
    weight.append(int(w_i))
    arr.append(S(n_i,int(h_i),int(w_i)))

arr.sort(key=lambda x: (x.h,-x.w))

for e in arr:
    print(e.n,e.h,e.w)